lib/streams.js

```diff
@@ -54,3 +54,77 @@ class WriteStream extends Writable {
 }
 
 exports.WriteStream = WriteStream
+
+class ByteStream extends Readable {
+  constructor (core, opts = {}) {
+    super()
+
+    this.core = core
+    this.byteOffset = opts.byteOffset || 0
+    this.byteLength = typeof opts.byteLength === 'number' ? opts.byteLength : -1
+    this.prefetch = typeof opts.prefetch === 'number' ? opts.prefetch : 32
+
+    this.index = 0
+    this.range = null
+  }
+
+  _open (cb) {
+    this._openp().then(cb, cb)
+  }
+
+  async _openp () {
+    if (this.byteLength === -1) {
+      await this.core.update()
+      this.byteLength = this.core.byteLength
+    }
+  }
+
+  async _read (cb) {
+    let data = null
+
+    if (this.byteLength === 0) {
+      this.push(null)
+      return cb(null)
+    }
+
+    let relativeOffset = 0
+
+    if (this.byteOffset > 0) {
+      const [block, byteOffset] = await this.core.seek(this.byteOffset)
+
+      this.byteOffset = 0
+      this.index = block
+
+      relativeOffset = byteOffset
+    }
+
+    this._prefetch(this.index + 1)
+    data = await this.core.get(this.index++)
```

Comment:
try catch this and return cb(err)

Reply:
No longer necessary to try-catch it due this?
https://github.com/holepunchto/hypercore/blob/e8f6a8f5dfb4d87e874605801183f20c950dc745/lib/streams.js#L76